config/v1/types_apiserver.go

```diff
@@ -0,0 +1,51 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// APIServer holds cluster-wide information about api-servers.  The canonical name is `cluster`
+type APIServer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              APIServerSpec   `json:"spec"`
+	Status            APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
+	// ServingCert is the TLS cert info for serving secure traffic.
+	ServingCerts APIServerServingCerts `json:"servingCerts,omitEmpty"`
+	// ClientCA is the certificate bundle for all the signers that you'll recognize for incoming client certificates
+	ClientCA ConfigMapNameReference `json:"clientCA,omitempty"`
+}
+
+type APIServerServingCerts struct {
+	// Default references a secret containing the default TLS cert info for serving secure traffic.
+	Default corev1.SecretReference `json:"default,omitempty"`
```

Comment:
add the funny +optional thing

Reply:
maybe `DefaultSecretName` ?

Reply:
(i think this needs a better name)

Reply:
> maybe `DefaultSecretName` ?

that doesn't provide more information

`DefaultServingCert` could work.

Reply:
@sanchezl I defer to @mfojtik 's request for a more specific name.  `DefaultServingCert`